src/query/functions/binary/common.go

```diff
@@ -93,16 +93,15 @@ func combineMetaAndSeriesMeta(
 	leftTags := meta.Tags
 	otherTags := otherMeta.Tags.TagMap()
 
-	metaTagsToAdd := make(models.Tags, 0)
-	otherMetaTagsToAdd := make(models.Tags, 0)
-	tags := make(models.Tags, 0)
+	metaTagsToAdd := make(models.Tags, 0, len(leftTags))
+	otherMetaTagsToAdd := make(models.Tags, 0, len(otherTags))
+	tags := models.EmptyTags()
```

Comment:
nit: may be better to initialize this with one of the other lengths, since there's a decent chance this will grow?
Can init it to initIndexSliceLength if you prefer

Reply:
Oh this are the ones to be added. Hard to say whether new common tags will always be greater than current series' unique tags. Even init length of initIndexSliceLength  = 10 is weird.

Reply:
If we're likely to be appending anything to it, should set some sort of non zero init value